File: app/src/main/java/com/sahibindencourseproject/adapter/WeatherItemAdapter.java
package com.sahibindencourseproject.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.sahibindencourseproject.R;
import com.sahibindencourseproject.api.model.WeatherItem;
import com.sahibindencourseproject.util.DateUtil;
import com.sahibindencourseproject.util.TemperatureUtil;

import java.util.ArrayList;

/**
 * Created by Ataer Caner on 2019-11-27.
 * Copyright (c) 2019 sahibinden. All rights reserved.
 */

public class WeatherItemAdapter extends RecyclerView.Adapter<WeatherItemAdapter.WeatherItemsViewHolder> {
    private ArrayList<WeatherItem> itemWeatherItem;
    private ItemClickListener itemClickListener;

    public interface ItemClickListener {
        void onItemClick(WeatherItem weatherItem);
    }

    class WeatherItemsViewHolder extends RecyclerView.ViewHolder {

        TextView txtDayName;
        TextView txtTemp;


        WeatherItemsViewHolder(View view) {
            super(view);

            txtDayName = view.findViewById(R.id.txtDayName);
            txtTemp = view.findViewById(R.id.txtTemp);

        }
    }


    public WeatherItemAdapter(ItemClickListener itemClickListener) {
        this.itemClickListener = itemClickListener;
        itemWeatherItem = new ArrayList<>();
    }


    @Override
    public WeatherItemsViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {

        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_weather, parent, false);


        return new WeatherItemsViewHolder(v);
    }


    @Override
    public void onBindViewHolder(WeatherItemsViewHolder holder, int position) {
        WeatherItem item = itemWeatherItem.get(position);
        String dayName = DateUtil.getGivenDayOfWeekAsName((position + DateUtil.getTodaysDayOfWeekAsIndex()) % 7);

        holder.txtDayName.setText(dayName);
        holder.txtTemp.setText(TemperatureUtil.getCelcius(item.getTemp().getDay()));
        holder.itemView.setOnClickListener(view -> itemClickListener.onItemClick(item));
    }

    @Override
    public int getItemCount() {
        return itemWeatherItem.size();
    }

    public void setItemWeatherItem(ArrayList<WeatherItem> itemWeatherItem) {
        this.itemWeatherItem = itemWeatherItem;
        notifyDataSetChanged();
    }

}

File: app/src/main/java/com/sahibindencourseproject/adapter/WeatherItemAdapter.kt
package com.sahibindencourseproject.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import androidx.recyclerview.widget.RecyclerView

import com.sahibindencourseproject.R
import com.sahibindencourseproject.api.model.WeatherItem
import com.sahibindencourseproject.databinding.ItemWeatherBinding
import com.sahibindencourseproject.util.DateUtil
import com.sahibindencourseproject.util.TemperatureUtil

import java.util.ArrayList

/**
 * Created by Ataer Caner on 2019-11-27.
 * Copyright (c) 2019 sahibinden. All rights reserved.
 */

class WeatherItemAdapter(private val itemClickListener: (WeatherItem) -> Unit) : RecyclerView.Adapter<WeatherItemAdapter.WeatherItemsViewHolder>() {
    private var itemWeatherItem: ArrayList<WeatherItem>? = null

    class WeatherItemsViewHolder(private val binding: ItemWeatherBinding): RecyclerView.ViewHolder(binding.root){

        fun bind(position: Int, item:WeatherItem, itemClickListener: (WeatherItem) -> Unit) {
            binding.position = position
            binding.weatherItem = item
            binding.root.setOnClickListener { itemClickListener.invoke(item) }
            binding.executePendingBindings()
        }
    }

    init {
        itemWeatherItem = ArrayList()
    }


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): WeatherItemsViewHolder {
        var binding = ItemWeatherBinding.inflate(LayoutInflater.from(parent.context),parent,false)

        return WeatherItemsViewHolder(binding)
    }


    override fun onBindViewHolder(holder: WeatherItemsViewHolder, position: Int) {
        val item = itemWeatherItem!![position]
        holder.bind(position,item,itemClickListener)
/*
        holder.tempBinding.position = position
        holder.tempBinding.weatherItem = item
        holder.tempBinding.root.setOnClickListener { itemClickListener.invoke(item) }
        holder.tempBinding.executePendingBindings()

 */
    }

    override fun getItemCount(): Int {
        return itemWeatherItem!!.size
    }

    fun setItemWeatherItem(itemWeatherItem: ArrayList<WeatherItem>) {
        this.itemWeatherItem = itemWeatherItem
        notifyDataSetChanged()
    }

}
